src/commands/createApiCenterService.ts

```diff
@@ -0,0 +1,95 @@
+        const serverRes = await apiCenterService.getSubServerList();
+        const locations = serverRes.resourceTypes.find((item: any) => item.resourceType === 'services')?.locations;
+        if (!locations || locations.length === 0) {
+            vscode.window.showWarningMessage(UiStrings.NoLocationAvailable);
+            return;
+        }
+        const location = await vscode.window.showQuickPick(locations, { placeHolder: UiStrings.SelectLocation });
+        if (!location) {
+            return;
+        }
+
+        await vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            title: UiStrings.CreateApiCenterService
+        }, async (progress, token) => {
+            progress.report({ message: UiStrings.GetResourceGroup });
+            const rgExist = await apiCenterService.isResourceGroupExist();
+            if (!rgExist) {
+                progress.report({ message: UiStrings.CreatingResourceGroup });
+                await AzureApiCenterService.confirmResourceGroupWithRetry(apiCenterService, location);
+            }
+
+            progress.report({ message: UiStrings.CreatingApiCenterService });
+            validateResponse(await apiCenterService.createOrUpdateApiCenterService(location));
+            // Retry mechanism to check API center creation status
+            await AzureApiCenterService.confirmServerStatusWithRetry(serverName, node, context);
+        });
+    };
+    export function validateResponse(response: any) {
+        if (response && (response as any).message) {
+            throw new Error((response as any).message);
+        }
+    };
+    export async function confirmResourceGroupWithRetry(apiCenterService: ApiCenterService, location: string) {
+        let retryCount = 0;
+        const maxRetries = 5;
+        const retryDelay = 10000; // 10 seconds
+        let result: ResourceGroup;
+        do {
+            validateResponse(result = await apiCenterService.createOrUpdateResourceGroup(location));
+            if (result && result.properties && result.properties.provisioningState && result.properties.provisioningState === 'Succeeded') {
+                break;
+            }
+            retryCount++;
+            await GeneralUtils.sleep(retryDelay);
+        } while (retryCount < maxRetries);
+        if (retryCount >= maxRetries) {
+            throw new Error(UiStrings.LongTimeToCreateResourceGroup);
+        }
+    };
+    export async function confirmServerStatusWithRetry(serverName: string, node: SubscriptionTreeItem, context: IActionContext) {
+        let retryCount = 0;
+        const maxRetries = 5;
+        const retryDelay = 10000; // 10 seconds
+        do {
+            try {
+                const resourceGraphService = new ResourceGraphService(node.subscription);
+                const apiCenter = await resourceGraphService.queryApiCenterByName(serverName);
+                if (apiCenter && apiCenter.length > 0) {
```

Comment:
why we need to refresh only when apiCenter.length > 0?

Reply:
the apicenter returns an array, if no such name apic created, will return [], we can confirm that this apic created and finded by graph api when length > 0